packages/lodestar/src/chain/chain.ts

```diff
@@ -78,6 +82,9 @@ export class BeaconChain implements IBeaconChain {
   readonly seenSyncCommitteeMessages = new SeenSyncCommitteeMessages();
   readonly seenContributionAndProof = new SeenContributionAndProof();
 
+  // caches computed proposers for epoch N+1 in the future
+  readonly nextEpochProposerDutyCache = new Map<Epoch, ValidatorIndex[]>();
```

Comment:
not sure if we need a separate class to manage this like the above caches and add a `prune(epoch)` method for it, leave it for @dapplion to consider

Reply:
I considered this also but wanted to keep it simple. But happy to create a separate class for it if @dapplion also agree thats preferred